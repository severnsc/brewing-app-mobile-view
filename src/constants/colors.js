export const primary = "#3183c8";
export const success = "#37c172";
export const danger = "#dc302f";
export const gray = "#adb5bd";
export const lightGray = "#e9ecee";
export const lighterGray = "#f1f3f5";
export const darkestGray = "#21252a";
