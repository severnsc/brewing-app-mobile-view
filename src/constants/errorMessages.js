export const INVALID_EMAIL = "Not a valid email!";
export const NON_UNIQUE_EMAIL = "Email is already taken! Try another email.";
export const EMPTY_USERNAME = "Username cannot be empty!";
export const NON_UNIQUE_USERNAME =
  "Username is already taken! Try another username.";
export const INVALID_PASSWORD = "Password must be at least 8 characters long!";
export const NETWORK_ERROR = "There was a problem with the network! Try again.";
export const NON_MATCHING_PASSWORD = "Must match the password!";
export const INVALID_LOGIN = "Invalid Username or Password!";
