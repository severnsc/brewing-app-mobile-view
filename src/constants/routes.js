export const HOME = "HOME";
export const APP = "APP";
export const STORYBOOK = "STORYBOOK";
export const CREATE_ACCOUNT = "CREATE_ACCOUNT";
export const LOGIN = "LOGIN";
