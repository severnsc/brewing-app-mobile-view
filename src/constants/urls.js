export const BACKEND_URL = "https://brewing-app-api.herokuapp.com";
