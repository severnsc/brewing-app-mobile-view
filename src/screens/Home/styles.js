import { StyleSheet } from "react-native";

export default StyleSheet.create({
	container: {
		flex: 1,
		justifyContent: "space-around",
		alignItems: "center",
		padding: 5
	},
	buttonRow: {
		width: "100%"
	},
	button: {
		marginTop: 10
	}
});
