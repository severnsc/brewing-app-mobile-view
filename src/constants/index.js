import { primary, success, danger, gray, lightGray, darkGray } from "./colors";
import iconSizes from "./iconSizes";
import { HOME, APP, STORYBOOK } from "./routes";
export {
	primary,
	success,
	danger,
	gray,
	lightGray,
	darkGray,
	iconSizes,
	HOME,
	APP,
	STORYBOOK
};
