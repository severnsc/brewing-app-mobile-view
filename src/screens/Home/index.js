import React from "react";
import { Title, Button } from "../../components";
import { Image, View } from "react-native";
import { LinearGradient } from "expo";
import styles from "./styles";
import { white } from "../../constants";

const Home = ({ navigation: { navigate } }) => (
	<LinearGradient colors={["#cffcf3", "#0ab892"]} style={{ flex: 1 }}>
		<View style={styles.container}>
			<Title value="Brewing App" color={white} />
			<Image source={require("../../../assets/icon.png")} />
			<View style={styles.buttonRow}>
				<Button
					onPress={() => navigate(CREATE_ACCOUNT)}
					value="Create an account"
					style={styles.button}
					secondary
				/>
				<Button
					onPress={() => navigate(LOGIN)}
					value="Login"
					style={styles.button}
					secondary
				/>
			</View>
		</View>
	</LinearGradient>
);

export default Home;
