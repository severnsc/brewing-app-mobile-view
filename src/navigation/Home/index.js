import { Home } from "../../screens";

const home = {
  HOME: {
    screen: Home,
    navigationOptions: { header: null, headerBackTitle: null }
  }
};

export default home;
