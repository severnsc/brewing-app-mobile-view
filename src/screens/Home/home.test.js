import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import Home from ".";
import { CREATE_ACCOUNT, LOGIN } from "../../constants";

describe("Home screen", () => {
	describe("layout", () => {
		it("should match the snapshot", () => {
			const home = renderer
				.create(<Home navigation={{ navigate: () => {} }} />)
				.toJSON();
			expect(home).toMatchSnapshot();
		});
	});
	describe("interactions", () => {
		describe("pressing the create account button", () => {
			it("should call the navigate prop with the CREATE_ACCOUNT constant", () => {
				const navigate = jest.fn();
				const navigation = { navigate };
				const home = shallow(<Home navigation={navigation} />);
				home
					.find("Button")
					.first()
					.simulate("press");
				expect(navigate).toBeCalledWith(CREATE_ACCOUNT);
			});
		});
		describe("pressing the login button", () => {
			it("should call the navigate prop with the LOGIN constant", () => {
				const navigate = jest.fn();
				const navigation = { navigate };
				const home = shallow(<Home navigation={navigation} />);
				home.find({ value: "Login" }).simulate("press");
				expect(navigate).toBeCalledWith(LOGIN);
			});
		});
	});
});
