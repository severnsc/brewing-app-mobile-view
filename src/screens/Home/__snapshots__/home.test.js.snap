// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home screen layout should match the snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "space-around",
      "padding": 5,
    }
  }
>
  <Text
    accessibilityLabel="Brewing App"
    style={
      Object {
        "color": "#fff",
        "fontSize": 40,
      }
    }
    testID="HomeText"
  >
    Brewing App
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../assets/icon.png",
      }
    }
    testID="Hero"
  />
  <View
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderColor": "#e9ecee",
          "borderWidth": 1,
          "marginTop": 10,
          "opacity": 1,
          "padding": 10,
        }
      }
      testID="ToCreateAccount"
    >
      <Text
        accessibilityLabel="Create an account"
        style={
          Object {
            "color": undefined,
            "fontSize": undefined,
          }
        }
      >
        Create an account
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderColor": "#e9ecee",
          "borderWidth": 1,
          "marginTop": 10,
          "opacity": 1,
          "padding": 10,
        }
      }
      testID="ToLogin"
    >
      <Text
        accessibilityLabel="Login"
        style={
          Object {
            "color": undefined,
            "fontSize": undefined,
          }
        }
      >
        Login
      </Text>
    </View>
  </View>
</View>
`;
